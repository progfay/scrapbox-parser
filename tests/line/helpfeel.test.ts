/* global describe it expect */

import '../jest-setup'

describe('helpfeel', () => {
  it('Simple helpfeel', () => {
    expect('? Simple helpfeel').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'helpfeel',
            text: 'Simple helpfeel'
          }
        ]
      }
    ])
  })

  it('No head `?` is not helpfeel', () => {
    expect('a ? not helpfeel').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'a ? not helpfeel'
          }
        ]
      }
    ])
  })

  it('Quoted ? is not helpfeel', () => {
    expect('> ? Quoted').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'quote',
            nodes: [
              {
                type: 'plain',
                text: ' ? Quoted'
              }
            ]
          }
        ]
      }
    ])
  })
})
