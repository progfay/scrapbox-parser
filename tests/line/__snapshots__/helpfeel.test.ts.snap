// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpfeel No head \`?\` is not helpfeel 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "raw": "a ? not helpfeel",
        "text": "a ? not helpfeel",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`helpfeel Quoted ? is not helpfeel 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "nodes": Array [
          Object {
            "raw": " ? Quoted",
            "text": " ? Quoted",
            "type": "plain",
          },
        ],
        "raw": "> ? Quoted",
        "type": "quote",
      },
    ],
    "type": "line",
  },
]
`;

exports[`helpfeel Simple helpfeel 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "raw": "? Simple helpfeel",
        "text": "Simple helpfeel",
        "type": "helpfeel",
      },
    ],
    "type": "line",
  },
]
`;
